js/viewer/viewer.js

```diff
@@ -188,7 +191,7 @@ $(document).ready(function() {
 		if (odfViewer.supportedMimes.indexOf(mimetype) !== -1 && $("#isPublic").val()){
 			// Single file public share, add button to allow view or edit
 			var button = document.createElement("a");
-			button.href = OC.generateUrl("apps/richdocuments/s/{token}?fileId={file_id}", { token: encodeURIComponent($("#sharingToken").val()), file_id: null });
+			button.href = OC.generateUrl("apps/richdocuments/public?fileId={file_id}&shareToken={shareToken}", { file_id: null, shareToken: encodeURIComponent($("#sharingToken").val()) });
```

Comment:
is this the kind of URL that people would be sending around ? what URL does the share dialog generate in this app ?

I'm only asking because changing the URL syntax could affect such users

Reply:
This is internal API used by Collabora App JavaScript to redirect to Editor instead of downloading.